scripts/microannotate_generator.py

```diff
@@ -25,7 +26,8 @@ def __init__(self, cache_root, repo_url, tokenize, remove_comments):
         self.repo_dir = os.path.join(cache_root, "mozilla-central")
 
     def generate(self):
-        repository.clone(self.repo_dir)
+        executor = concurrent.futures.ThreadPoolExecutor(max_workers=os.cpu_count() + 1)
```

Comment:
You shoud use the executor as a context manager to make sure the threads are cleaned up at the end.